test/node/connection-handshake.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const test = require('tap').test;
+
+const jstp = require('../..');
+
+const app = require('../fixtures/application');
+const application = new jstp.Application(app.name, app.interfaces);
+
+const Transport = require('../unit/mock/transport');
+
+test('must perform an anonymous handshake', (test) => {
+  test.plan(3);
+
+  const port = 3333;
+  const server = jstp.tcp.createServer(port, [application], app.authCallback);
+  server.listen(() => {
+    const client = jstp.tcp.createClient({ host: 'localhost', port });
+
+    client.connectAndHandshake(app.name, null, null, (error, connection) => {
+      test.assertNot(error, 'must be no error');
+      test.equal(connection.username, null, 'username must be null');
+      test.equal(connection.sessionId, app.sessionId,
+        'session id must be equal to the one provided by authCallback');
+      client.disconnect();
+      server.close();
+    });
+  });
+});
+
+
+test('must perform a handshake with credentials', (test) => {
+  test.plan(3);
+
+  const port = 3334;
+  const server = jstp.tcp.createServer(port, [application], app.authCallback);
+  server.listen(() => {
+    const client = jstp.tcp.createClient({ host: 'localhost', port });
+
+    client.connectAndHandshake(app.name, app.login, app.password,
+      (error, connection) => {
+        test.assertNot(error, 'must be no error');
+        test.equal(connection.username, app.login,
+          'username must be same as the one passed with handshake');
+        test.equal(connection.sessionId, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+        client.disconnect();
+        server.close();
+      }
+    );
+  });
+});
+
+test('must not perform a handshake with invalid credentials', (test) => {
+  test.plan(2);
+
+  const port = 3335;
+  const server = jstp.tcp.createServer(port, [application], app.authCallback);
+  server.listen(() => {
+    const client = jstp.tcp.createClient({ host: 'localhost', port });
+
+    client.connectAndHandshake(app.name, app.login, 'psssword', (error) => {
```

Comment:
I think it's better to change 'psssword' to something more explicitly wrong, as now it is very similar to the correct one.